fix: enhance account selection logic in AccountSelectorActions

sidmorizon opened the pull request:
- Introduced a mutex for auto-selecting the next account to prevent race conditions.
- Added conditions to handle specific wallet transitions and ensure proper account focus.
- Refactored account selection logic to improve clarity and maintainability.
- Updated the handling of others wallet accounts to ensure correct selection and focus behavior.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved reliability of automatic account selection across HD/Hardware and Other wallets.
  * Corrected cases where the wrong wallet/account could be focused after removing wallets or accounts.
  * Reduced inconsistent states caused by overlapping auto-select actions.

* **Refactor**
  * Streamlined the auto-select flow to handle more scenarios and ensure consistent focus and selection behavior.

* **Chores**
  * Minor code organization updates with no user-visible changes.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sidmorizon committed:
fix: enhance account selection logic in AccountSelectorActions

- Introduced a mutex for auto-selecting the next account to prevent race conditions.
- Added conditions to handle specific wallet transitions and ensure proper account focus.
- Refactored account selection logic to improve clarity and maintainability.
- Updated the handling of others wallet accounts to ensure correct selection and focus behavior.

revan-zhang commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/1fbc52da-4db8-4996-bf60-91bfca2fccc8?refs=a64826112690286506cce3118123c8ef49cb8fbe&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/1fbc52da-4db8-4996-bf60-91bfca2fccc8/license?refs=a64826112690286506cce3118123c8ef49cb8fbe&source=prChecksComment)

sidmorizon committed:
fix: comment out debugger statements in AccountSelectorActions

- Removed active debugger statements to clean up the code and prevent unintended breakpoints during execution.
- Maintained the logic for account selection without altering functionality.

originalix commented on packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:
```diff
@@ -356,6 +356,20 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
           // debugger;
         }
 
+        if (
+          oldSelectedAccount.walletId === 'hd-2' &&
+          newSelectedAccount.walletId === 'hd-4'
+        ) {
+          // debugger;
+        }
+
+        if (
+          oldSelectedAccount.walletId === undefined &&
+          newSelectedAccount.walletId === 'hd-2'
+        ) {
+          // debugger;
+        }
```
这里移除下

huhuanming replied:
需要移除么？

huhuanming replied:
😳

sidmorizon committed:
fix: remove unnecessary walletId checks in AccountSelectorActions

- Commented out redundant checks for specific wallet transitions to streamline account selection logic.
- Maintained existing functionality while improving code clarity and maintainability.
